book/src/appendices/b_migration_notes/cgmath_to_nalgebra.md

```diff
@@ -0,0 +1,245 @@
+# CG Math to NAlgebra
+
+## How To Use
+
+This cheat sheet is split up into the following sections:
+
+* **Type Changes:** Previously you used `this::Type`, now you use `another::Thing`
+* **Logic Changes:** Previously you had `object.method(args)`, now you use `object.other(stuff)`
+
+Most changes will have accompanying explanations and code examples on how to switch.
+
+This document is by no means exhaustive, so if there is something missing, or if you can clarify any changes, please correct this!
+
+The text is designed to be searchable, so if you are looking for a specific type or method, please use Ctrl + F: `TypeName`. If you cannot find it in the document, likely we missed it during writing. Please send us a pull request!
+
+## Type Changes
+
+Many types retain the same type name, just under the `nalgebra` namespace:
+
+```patch
+-use amethyst::core::cgmath::{Vector2, Vector3, Matrix4};
++use amethyst::core::nalgebra::{Vector2, Vector3, Matrix4};
+```
+
+We will not list the names of every type with the same simple name, but will try to list the changes for types whose simple names are different:
+
+```patch
+-cgmath::Ortho
++nalgebra::Orthographic3
+
+-cgmath::PerspectiveFov
++nalgebra::Perspective3
+```
+
+## Logic Changes
+
+* `cgmath` to `nalgebra` functions:
+
+    ```patch
+    -Vector3::unit_z()
+    +Vector3::z()
+
+    -matrix4.z.truncate()
+    +matrix4.column(2).xyz().into()
+
+    -matrix4.transform_point(origin)
+    +matrix4.transform_point(&origin)
+    ```
+
+* `amethyst::core::transform::Transform`
+
+    - Transformation values are accessed / mutated through accessor methods.
+
+        ```patch
+        -transform.translation = Vector3::new(5.0, 2.0, -0.5);
+        -transform.scale = Vector3::new(2.0, 2.0, 2.0);
+        -transform.rotation = Quaternion::new(1.0, 0.0, 0.0, 0.0);
+        +transform.set_xyz(5.0, 2.0, -0.5);
+        +transform.set_scale(2.0, 2.0, 2.0);
+        +transform.set_rotation(Unit::new_normalize(Quaternion::new(1.0, 0.0, 0.0, 0.0)));
+
+        // Translations
+        -transform.translation = Vector3::new(0.0, 0.0, 0.0);
+        +*transform.translation_mut() = Vector3::new(0.0, 0.0, 0.0);
+
+        -transform_0.translation - transform_1.translation
+        +transform_0.translation() - transform_1.translation()
+
+        -transform.translation[0] = x;
+        +transform.set_x(position.x);
+
+        -translation.x += 0.1;
+        -translation.y -= 0.1;
+        +translation.translate_x(0.1);
+        +translation.translate_y(-0.1);
```

Comment:
You are correct and I think this is a typo. Is there any reason in particular why you prefer that way of writing it though? I added the new translation methods mostly so that people wouldn't have to write that entire line :sweat_smile:

Reply:
oops, shall fix the typo and add the alternative as well